micro_manager/micro_manager.py

```diff
@@ -551,12 +580,48 @@ def _solve_micro_simulations(self, micro_sims_input: list) -> list:
         micro_sims_output = [None] * self._local_number_of_sims
 
         for count, sim in enumerate(self._micro_sims):
-            start_time = time.time()
-            micro_sims_output[count] = sim.solve(micro_sims_input[count], self._dt)
-            end_time = time.time()
+            # If micro simulation has not crashed in a previous iteration, attempt to solve it
+            if not self._has_sim_crashed[count]:
+                # Attempt to solve the micro simulation
+                try:
+                    start_time = time.time()
+                    micro_sims_output[count] = sim.solve(
+                        micro_sims_input[count], self._dt
+                    )
+                    end_time = time.time()
+                    # Write solve time of the macro simulation if required and the simulation has not crashed
+                    if self._is_micro_solve_time_required:
+                        micro_sims_output[count]["micro_sim_time"] = (
+                            end_time - start_time
+                        )
 
-            if self._is_micro_solve_time_required:
-                micro_sims_output[count]["micro_sim_time"] = end_time - start_time
+                # If simulation crashes, log the error and keep the output constant at the previous iteration's output
+                except Exception as error_message:
+                    self._logger.error(
+                        "Micro simulation at macro coordinates {} has experienced an error. "
+                        "See next entry on this rank for error message.".format(
+                            self._mesh_vertex_coords[count]
+                        )
+                    )
+                    self._logger.error(error_message)
+                    self._has_sim_crashed[count] = True
+
+        # Interpolate result for crashed simulation
+        if self._has_sim_crashed.count(True) > 0:
```

Comment:
Is this condition a must? If there are no crashed simulations, I would expect the logic in the next step to yield `unset_sims = []`.

Reply:
No, it is redundant.